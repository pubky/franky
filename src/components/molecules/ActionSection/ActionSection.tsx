import { ReactNode } from 'react';

import * as Atoms from '@/atoms';
import * as Libs from '@/libs';

interface ActionSectionProps {
  children?: ReactNode;
  className?: string;
  actions?: Array<{
    label: string;
    icon?: ReactNode;
    onClick: () => void;
    variant?: 'default' | 'secondary' | 'outline' | 'ghost' | 'brand';
  }>;
}

export function ActionSection({ children, className, actions = [] }: ActionSectionProps) {
  return (
    <Atoms.Container className={Libs.cn('gap-6 w-full flex-row', className)}>
      <Atoms.Container className="flex-col items-start gap-3 justify-start w-full">{children}</Atoms.Container>
      {actions.length > 0 && (
        <Atoms.Container className="flex-row gap-3">
          {actions.map((action, index) => (
            <Atoms.Button
              key={index}
              variant={action.variant || 'secondary'}
              className="rounded-full"
              onClick={action.onClick}
            >
              {action.icon}
              {action.label}
            </Atoms.Button>
          ))}
        </Atoms.Container>
      )}
    </Atoms.Container>
  );
}
