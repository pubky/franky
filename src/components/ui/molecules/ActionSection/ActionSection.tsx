import { ReactNode } from 'react';
import { Button, Container } from '@/components/ui';
import { cn } from '@/libs';

interface ActionSectionProps {
  children?: ReactNode;
  className?: string;
  actions?: Array<{
    label: string;
    icon?: ReactNode;
    onClick: () => void;
    variant?: 'default' | 'secondary' | 'outline' | 'ghost' | 'brand';
  }>;
}

export function ActionSection({ children, className, actions = [] }: ActionSectionProps) {
  return (
    <Container className={cn('gap-6', className)}>
      {children}
      {actions.length > 0 && (
        <Container className="flex-row gap-3">
          {actions.map((action, index) => (
            <Button
              key={index}
              variant={action.variant || 'secondary'}
              className="rounded-full"
              onClick={action.onClick}
            >
              {action.icon}
              {action.label}
            </Button>
          ))}
        </Container>
      )}
    </Container>
  );
}
