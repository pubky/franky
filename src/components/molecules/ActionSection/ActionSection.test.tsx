import { describe, it, expect, vi } from 'vitest';
import { render, screen } from '@testing-library/react';
import { ActionSection } from './ActionSection';

// Mock UI components
vi.mock('@/components/ui', () => ({
  Container: ({ children, className }: { children: React.ReactNode; className?: string }) => (
    <div data-testid="container" className={className}>
      {children}
    </div>
  ),
  Button: ({
    children,
    variant,
    className,
    onClick,
  }: {
    children: React.ReactNode;
    variant?: string;
    className?: string;
    onClick?: () => void;
  }) => (
    <button data-testid={`button-${variant || 'default'}`} className={className} onClick={onClick}>
      {children}
    </button>
  ),
}));

describe('ActionSection', () => {
  it('renders with actions', () => {
    const mockAction = vi.fn();
    const actions = [
      { label: 'Continue', onClick: mockAction, variant: 'default' as const },
      { label: 'Back', onClick: mockAction, variant: 'outline' as const },
    ];

    render(<ActionSection actions={actions} />);

    const continueButton = screen.getByTestId('button-default');
    const backButton = screen.getByTestId('button-outline');

    expect(continueButton).toBeInTheDocument();
    expect(continueButton).toHaveTextContent('Continue');
    expect(backButton).toBeInTheDocument();
    expect(backButton).toHaveTextContent('Back');
  });
});

describe('ActionSection - Snapshots', () => {
  it('matches snapshot with default props', () => {
    const { container } = render(<ActionSection />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('matches snapshot with custom className', () => {
    const { container } = render(<ActionSection className="custom-actions" />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('matches snapshot with children', () => {
    const { container } = render(
      <ActionSection>
        <div>Custom content</div>
      </ActionSection>,
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('matches snapshot with single action', () => {
    const mockAction = vi.fn();

    const { container } = render(<ActionSection actions={[{ label: 'Single Action', onClick: mockAction }]} />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('matches snapshot with multiple actions', () => {
    const mockAction = vi.fn();

    const { container } = render(
      <ActionSection
        actions={[
          { label: 'Continue', onClick: mockAction, variant: 'default' as const },
          { label: 'Back', onClick: mockAction, variant: 'outline' as const },
        ]}
      />,
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('matches snapshot with action containing icon', () => {
    const mockAction = vi.fn();

    const { container } = render(
      <ActionSection
        actions={[
          {
            label: 'Action with Icon',
            icon: <span>ðŸš€</span>,
            onClick: mockAction,
          },
        ]}
      />,
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
