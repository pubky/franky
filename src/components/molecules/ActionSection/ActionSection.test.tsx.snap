// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ActionSection - Snapshots > matches snapshot with action containing icon 1`] = `
<div
  class="mx-auto flex gap-6 w-full flex-row"
  data-testid="container"
>
  <div
    class="mx-auto flex flex-col items-start gap-4 justify-start w-full"
    data-testid="container"
  />
  <div
    class="mx-auto w-full flex flex-row gap-3"
    data-testid="container"
  >
    <button
      class="inline-flex items-center justify-center whitespace-nowrap text-sm transition-all disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg:not([class*=\\"size-\\"])]:size-4 shrink-0 [&_svg]:shrink-0 outline-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/40 aria-invalid:border-destructive font-semibold cursor-pointer border shadow-xs bg-secondary text-secondary-foreground hover:bg-secondary/80 h-10 gap-1 px-4 py-2 has-[>svg]:px-4 rounded-full"
      data-slot="button"
      data-variant="secondary"
    >
      <span>
        ðŸš€
      </span>
      Action with Icon
    </button>
  </div>
</div>
`;

exports[`ActionSection - Snapshots > matches snapshot with children 1`] = `
<div
  class="mx-auto flex gap-6 w-full flex-row"
  data-testid="container"
>
  <div
    class="mx-auto flex flex-col items-start gap-4 justify-start w-full"
    data-testid="container"
  >
    <div>
      Custom content
    </div>
  </div>
</div>
`;

exports[`ActionSection - Snapshots > matches snapshot with custom className 1`] = `
<div
  class="mx-auto flex gap-6 w-full flex-row custom-actions"
  data-testid="container"
>
  <div
    class="mx-auto flex flex-col items-start gap-4 justify-start w-full"
    data-testid="container"
  />
</div>
`;

exports[`ActionSection - Snapshots > matches snapshot with default props 1`] = `
<div
  class="mx-auto flex gap-6 w-full flex-row"
  data-testid="container"
>
  <div
    class="mx-auto flex flex-col items-start gap-4 justify-start w-full"
    data-testid="container"
  />
</div>
`;

exports[`ActionSection - Snapshots > matches snapshot with multiple actions 1`] = `
<div
  class="mx-auto flex gap-6 w-full flex-row"
  data-testid="container"
>
  <div
    class="mx-auto flex flex-col items-start gap-4 justify-start w-full"
    data-testid="container"
  />
  <div
    class="mx-auto w-full flex flex-row gap-3"
    data-testid="container"
  >
    <button
      class="inline-flex items-center justify-center whitespace-nowrap text-sm transition-all disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg:not([class*=\\"size-\\"])]:size-4 shrink-0 [&_svg]:shrink-0 outline-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/40 aria-invalid:border-destructive font-semibold cursor-pointer border shadow-xs bg-brand/16 text-brand hover:!bg-brand/30 border-brand h-10 gap-1 px-4 py-2 has-[>svg]:px-4 rounded-full"
      data-slot="button"
      data-variant="default"
    >
      Continue
    </button>
    <button
      class="inline-flex items-center justify-center whitespace-nowrap text-sm transition-all disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg:not([class*=\\"size-\\"])]:size-4 shrink-0 [&_svg]:shrink-0 outline-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/40 aria-invalid:border-destructive font-semibold cursor-pointer border shadow-xs hover:text-accent-foreground bg-input/30 border-input hover:bg-input/50 h-10 gap-1 px-4 py-2 has-[>svg]:px-4 rounded-full"
      data-slot="button"
      data-variant="outline"
    >
      Back
    </button>
  </div>
</div>
`;

exports[`ActionSection - Snapshots > matches snapshot with single action 1`] = `
<div
  class="mx-auto flex gap-6 w-full flex-row"
  data-testid="container"
>
  <div
    class="mx-auto flex flex-col items-start gap-4 justify-start w-full"
    data-testid="container"
  />
  <div
    class="mx-auto w-full flex flex-row gap-3"
    data-testid="container"
  >
    <button
      class="inline-flex items-center justify-center whitespace-nowrap text-sm transition-all disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg:not([class*=\\"size-\\"])]:size-4 shrink-0 [&_svg]:shrink-0 outline-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/40 aria-invalid:border-destructive font-semibold cursor-pointer border shadow-xs bg-secondary text-secondary-foreground hover:bg-secondary/80 h-10 gap-1 px-4 py-2 has-[>svg]:px-4 rounded-full"
      data-slot="button"
      data-variant="secondary"
    >
      Single Action
    </button>
  </div>
</div>
`;
